src/charm.py

```diff
@@ -151,11 +158,11 @@ def _process_wheel_file(self, event):  # noqa: C901
 
                 module_name = out.split("\n")
                 if self.unit.is_leader():
-                    self._state.module_name = module_name[0].split("/")[-1]
+                    self._state.module_name = module_name[0].rsplit("/", maxsplit=1)
```

Comment:
As previous comment, `rsplit` returns a list.
You could use `pathlib` for this as well.
Also, would it make sense to do some error handling on `module_name`?

Reply:
It probably isn't necessary as the wheel file would not have built properly without a valid module name.